### PR TITLE
Upgrade alibi-detect component to use alibi-detect 0.4.2

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

Upgrades alibi-detect server to use `alibi-detect=0.4.2`.

This is separate from https://github.com/SeldonIO/seldon-core/pull/2404 to only include `alibi-detect` server changes.

This also splits image build process to independently handle `alibi-explain` and `alibi-detect` servers.

**Which issue(s) this PR fixes**:

Closes https://github.com/SeldonIO/seldon-core/issues/2403

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
updates alibi-detect to version 0.4.2
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Sep 14 10:44:21 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=2

**Comment by seldondev** [comment]
Mon Sep 14 10:44:34 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=1

**Comment by seldondev** [comment]
Mon Sep 14 10:46:49 UTC 2020
The logs for [integration] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=3

File: testing/scripts/kind_test_all.sh [hunk 1]
```diff
@@ -125,18 +125,32 @@ if [[ ${KIND_EXIT_VALUE} -eq 0 ]]; then
             echo "SKIPPING PREPACKAGED IMAGE BUILD..."
         fi
 
+        echo "Files changed in alibi explain folder:"
+        git --no-pager diff --exit-code --name-only origin/master ../../components/alibi-explain-server/
+        ALIBI_MODIFIED=$?
+        if [[ $ALIBI_MODIFIED -gt 0 ]]; then
```
Comment (axsaucedo) [inline testing/scripts/kind_test_all.sh:131]: You should change this to ALIBI_EXPLAIN_MODIFIED and same below so they are separate (and then add them to the values at the bottom)

Comment (RafalSkolasinski) [reply testing/scripts/kind_test_all.sh]: done

**Comment by seldondev** [comment]
Mon Sep 14 17:48:24 UTC 2020
The logs for [pr-build] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=4

**Comment by seldondev** [comment]
Mon Sep 14 17:48:24 UTC 2020
The logs for [lint] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=5

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2431#issuecomment-692219031 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-692219031/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2431#issuecomment-692557902 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-692557902/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Sep 15 08:58:45 UTC 2020
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=7

**Comment by seldondev** [comment]
Tue Sep 15 08:58:47 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2431/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2431 --build=6

**Pull request merged**